### PR TITLE
Added support for numeric spaces for LocalGridSearcher

### PR DESCRIPTION
**Pull request by gradientsky**
*Description of changes:*
Added support for numeric spaces for LocalGridSearcher

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-1448-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1448/1/index.html

**Comment by szha** [comment]
Job PR-1448-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1448/2/index.html

File: core/src/autogluon/core/searcher/local_grid_searcher.py [hunk 1]
```diff
@@ -13,10 +15,21 @@
...
+        """
+        Parameters
+        ----------
+        grid_numeric_spaces_points_number: int, default = 4
+            number of data point to sample from numeric space
+        grid_num_sample_settings: dict (optional), default = None
+            mapping between numeric space name and number of points to sample from it. Example {'a': 4} means
+            sample 4 points from space 'a'. If no value present in this map, then `grid_numeric_spaces_points_number` will be used.
+        """
+        super(LocalGridSearcher, self).__init__(**kwargs)
```
Comment (Innixma) [inline core/src/autogluon/core/searcher/local_grid_searcher.py:30]: nit: why not just `super().__init__(**kwargs)`

File: core/src/autogluon/core/searcher/local_grid_searcher.py [hunk 2]
```diff
@@ -26,12 +39,27 @@ def _get_params_space(self) -> dict:
...
         for key, val in self.search_space.items():
             if isinstance(val, Space):
-                if not isinstance(val, Categorical):
+                samples_num = self._get_samples_number(key)
+                if isinstance(val, Int):
+                    samples = min(val.upper - val.lower + 1, samples_num)
+                    param_space[key] = np.linspace(val.lower, val.upper, samples, dtype=int)
+                elif isinstance(val, Real):
+                    space = np.geomspace if val.log else np.linspace
+                    param_space[key] = space(val.lower, val.upper, num=samples_num)
```
Comment (Innixma) [inline core/src/autogluon/core/searcher/local_grid_searcher.py:48]: idk if its worth addressing but now this searcher doesn't respect default values on the first iteration. Worth mentioning in documentation that grid with Int/Real ignores 'default' values in search spaces.

**Review by Innixma — APPROVED** [review]
LGTM, added minor comments

File: core/src/autogluon/core/searcher/local_grid_searcher.py (hunk 1 above)
Comment (gradientsky) [reply core/src/autogluon/core/searcher/local_grid_searcher.py:30]: fixed

File: core/src/autogluon/core/searcher/local_grid_searcher.py (hunk 2 above)
Comment (gradientsky) [reply core/src/autogluon/core/searcher/local_grid_searcher.py:48]: Added comment

**Comment by szha** [comment]
Job PR-1448-4 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1448/4/index.html

**Comment by szha** [comment]
Job PR-1448-7 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-1448/7/index.html

**Pull request merged**